s4e-web/src/app/views/map-view/state/common.model.ts

```diff
@@ -4,4 +4,5 @@ export interface IUILayer {
   cid: ID;
   caption: string;
   active: boolean;
+  favourite?: boolean;
```

Comment:
Settling for field which can not be undefined is a better option. And just setting it always to false in `overlay.model.ts`

```
export function convertToUIOverlay(overlay: Overlay, geoServerUrl: string, active: boolean = false): UIOverlay {
  return {
    ...overlay,
    olLayer: new Image({
      source: new ImageWMS({
        crossOrigin: 'Anonymous',
        url: geoServerUrl,
        serverType: 'geoserver',
        params: {LAYERS: overlay.id}
      })
    }),
    cid: overlay.id,
    active: active,
    favourite: false
  };
}
```

Reply:
It's fine with me as it's our use case, although in yesterday conversation you mention that favourite will not be included in all converted items. Example Products can be favorite, but layers won't be